concordia/static/js/asset-reservation.js

```diff
@@ -1,5 +1,28 @@
-/* global jQuery displayMessage buildErrorMessage */
-/* exported attemptToReserveAsset */
+/* global jQuery displayMessage displayHtmlMessage buildErrorMessage */
+/* exported attemptToReserveAsset checkForReviewers */
+
+function addUnloadListener(reservationURL) {
+    window.addEventListener('beforeunload', function() {
+        var payload = {
+            release: true,
+            csrfmiddlewaretoken: jQuery(
+                'input[name="csrfmiddlewaretoken"]'
+            ).val()
+        };
+
+        // We'll try Beacon since that's reliable but until we can drop support for IE11 we need a fallback:
+        if ('sendBeacon' in navigator) {
+            navigator.sendBeacon(
+                reservationURL,
+                new Blob([jQuery.param(payload)], {
+                    type: 'application/x-www-form-urlencoded'
+                })
+            );
+        } else {
+            jQuery.ajax({url: reservationURL, type: 'POST', data: payload});
```

Comment:
We might want to weight the importance of the user getting a response faster vs. getting the lock released by having the AJAX branch trigger the cancellation prompt (see https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload#Example).